### PR TITLE
CLI improvements

### PR DESCRIPTION
**Pull request by lieut-data**
There's a huge swath of fixes and improvements I found necessary to test the public channels materialization feature: https://github.com/mattermost/mattermost-server/pull/9349.

* ltops: dump `--help` automatically if no flags passed to subcommands
* ltops: improved logging, including routing loadtest logs through logrus for easier human consumption, while including continuing to dump logs to the output file
* ltops: require `--cluster` consistently (vs. `--name` or `--cluster` or an arg sometimes)
* ltops: sort flags by definition for greater `--help` clarity
* ltops deploy: allow only mattermost or loadtest (or both) to be deployed (for faster iteration)
* ltops deploy: delay completion until the corresponding resources have actually rebooted
* ltops deploy: improve parallelization (terraform only)
* ltops deploy: in addition to files and urls, resolve PR#s, releases and `master` for easier deployment
* ltops loadtest distinguish results by HTTP method in addition to endpoint, and tweak rendering style for channel names, ids, etc.
* ltops loadtest: dump timing results every 100 results to avoid lost data if the test is interrupted and to help in showing progress, and update ltparse to aggregate back into a single result set
* ltops loadtest: fix a race condition in setting `Stderr` to `Stdout` for ssh sessions: `bytes.Buffer` isn't safe for this use case. This eliminates the `\u0000` seen in bulkloading logs from the server.
* ltops loadtest: fix a regression with the introduction of private channels that sometimes broke entity calls
* ltops loadtest: install dot and graphviz to actually capture pprof traces (terraform only)
* ltops loadtest: send the bulkload file in chunks to improve visibility into this slow process
* ltops loadtest: support `--config alternate-loadtestconfig.json` to easily tweak loadtest parameters (terraform only)
* ltops loadtest: use instances of `rand.Rand` vs. the global `rand` to aid in deterministic bulkload generation. Entities are still seeded randomly.
* ltops loadtest: work around `mattermost import bulk` limitations with long lines by artificially splitting up the generated file. This is primarily a problem once the number of users and channels increases.
* ltops db: new command to spawn `mysql` to the database in question
* ltops ssh: accept command to run: `ltops ssh --cluster jesse-hallam-loadtest-base cat /etc/hosts`
* ltops ssh: default to instance `0` when ssh-ing, and require `--instance-id` otherwise
* ltops destroy: an alias of `ltops delete` (helps with the terraform mindset) 
* ltparse: fix a rendering bug when the baseline didn't have an endpoint
* remove a bunch of confusing or commented code

### CODE REVIEW

File: cmd/ltops/db.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	"github.com/mattermost/mattermost-load-test/ltops"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var dbCommand = &cobra.Command{
+	Use:     "db",
+	Short:   "Launches mysql connected to the cluster database",
```
Comment (jwilander) [inline cmd/ltops/db.go:18]: 🎉

File: cmd/ltops/db.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+			return err
+		}
+
+		cluster, err := LoadCluster(filepath.Join(workingDir, clusterName))
+		if err != nil {
+			return errors.Wrap(err, "couldn't load cluster")
+		}
+
+		settings, err := cluster.DBSettings()
+		if err != nil {
```
Comment (jwilander) [inline cmd/ltops/db.go:34]: I'm assuming this just errors out for k8s cluster? Might be helpful to add to the help text that only terraform is supported right now

File: cmd/ltops/loadtest.go [hunk 3]
```diff
@@ -3,26 +3,29 @@ package main
...
-	Short: "Runs a mattermost-load-test command against the given cluster",
+	Use:     "loadtest",
+	Short:   "Runs a mattermost-load-test command against the given cluster",
+	PreRunE: showHelpIfNoFlags,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		loadtestOptions := &ltops.LoadTestOptions{}
 		clusterName, _ := cmd.Flags().GetString("cluster")
+
+		loadtestOptions := &ltops.LoadTestOptions{}
+		loadtestOptions.ConfigFile, _ = cmd.Flags().GetString("config")
```
Comment (jwilander) [inline cmd/ltops/loadtest.go:21]: 🎉

**Review by jwilander — APPROVED** [review]
This is awesome, tons of great improvements here 🎉 🎉

File: cmd/ltops/db.go (hunk 2 above)
Comment (lieut-data) [reply cmd/ltops/db.go:34]: I've exposed the `DBSettings` within the k8s cluster, but I'm uncertain if it actually works without modification: is something special needed to expose the database outside the k8s cluster?

**Review by crspeller — APPROVED** [review]

**Pull request merged**